shared/utility/math/filter/ParticleFilter.hpp

```diff
@@ -26,8 +26,42 @@
 #include <random>
 #include <vector>
 
+#include "utility/math/stats/multivariate.hpp"
+#include "utility/math/stats/resample/multinomial.hpp"
+#include "utility/math/stats/resample/resample.hpp"
+#include "utility/math/stats/resample/residual.hpp"
+#include "utility/math/stats/resample/stratified.hpp"
+#include "utility/math/stats/resample/systematic.hpp"
+
 namespace utility::math::filter {
 
+    using utility::math::stats::MultivariateNormal;
+
+    // The possible resampling methods
+    // See http://users.isy.liu.se/rt/schon/Publications/HolSG2006.pdf for an explanation on each one
+    enum class ResampleMethod { MULTINOMIAL, RESIDUAL, STRATIFIED, SYSTEMATIC };
+
+    /**
+     * @author Alex Biddulph, Trent Houliston
+     *
+     * @details Uses a set of particles (or samples) to represent possible states in a system. Each particle is randomly
+     * selected from a multivariate normal distribution. Each particle is weighted by how likely it is to be the real
+     * state of the system. Particles with a higher weight are more likely to be resampled (with noise) during the time
+     * update.
+     *
+     * Time Update: Resamples all particles, perturbing the resampled particles with the model's process noise, and asks
+     * the model to predict what should happen to each particle after an elapsed amount of time. Also known as the
+     * predict step.
```

Comment:
Is this really the predict step?

Reply:
According to some matlab docs I read, what they call the predict step sounds like our time update step. But if you can word this whole thing better I am open to suggestions

Reply:
I mean I don't have a problem with the naming.. kinda makes sense

Reply:
It's just a pain that our measurement function is called "predict" lol

Reply:
It is, I can remove that part of the comment if you think it will lead to confusion?

Reply:
Yeah I would just to avoid future confusion